### PR TITLE
No follow

### PR DESCRIPTION
**Pull request by alexcb**
This introduces a `--no-follow` flag for both COPY and SAVE ARTIFACT commands, which pass the NoFollow flag to buildkit's CopyOp which will prevent symlinks from being followed. This allows users to save and copy invalid symlinks (which point to non-existent paths).

When using `SAVE ARTIFACT` on files, `--no-follow` is required if the file is an invalid symlink. Furthermore, when that file is copied `--no-follow` must also be used on the `COPY` command.

However, when using `SAVE ARTIFACT` on directories which contain one or more symlink, the `--no-follow` flag has no effect on buildkit's internal workings, and therefore is not required by earthly. The `--no-follow` flag is still however required on the `COPY` command.

### CODE REVIEW

**Comment by vladaionescu** [comment]
I would maybe call this `--symlink-no-follow` so that it's obvious to the reader what "no follow" refers to.

**Comment by alexcb** [comment]
> so that it's obvious to the reader what "no follow" refers to.

That's a good point, it'll make it easier to search for.

File: earthfile2llb/converter.go [hunk 1]
```diff
@@ -496,7 +497,7 @@ func (c *Converter) RunExitCode(ctx context.Context, commandName string, args, m
 		"%s %s%s%s",
 		commandName, // eg IF
 		strIf(privileged, "--privileged "),
-		strIf(noCache, "--no-cache "),
+		strIf(noCache, "--symlink-no-cache "),
```
Comment (vladaionescu) [inline earthfile2llb/converter.go:500]: Probably unintentional? :-)

File: earthfile2llb/converter.go [hunk 2]
```diff
@@ -618,7 +619,7 @@ func (c *Converter) Run(ctx context.Context, args, mounts, secretKeyValues []str
 		strIf(withDocker, "--with-docker "),
 		strIf(withEntrypoint, "--entrypoint "),
 		strIf(pushFlag, "--push "),
-		strIf(noCache, "--no-cache "),
+		strIf(noCache, "--symlink-no-cache "),
```
Comment (vladaionescu) [inline earthfile2llb/converter.go:622]: Same here

File: earthfile2llb/converter.go (hunk 1 above)
Comment (alexcb) [reply earthfile2llb/converter.go]: ouch, that was a bad find-and-replace. Good catch.

**Review by vladaionescu — APPROVED** [review]

**Comment by sdaschner** [comment]
Stumbled upon this only by accident (which luckily solved my issue).

Can this be added to the documentation? 🙂

**Pull request merged**